### DISCUSSION TITLE
module-alias is ignored for macros

### DISCUSSION THREAD

**Discussion by bouncyboro** [lfe/lfe#495] [Q&A] [answered]
If macro is exported and used inside another module by aliased module name (`module-alias` in `defmodule` to rename the module with the macro definition), LFE compiler assumes it is a function and Erlang complains that undefined functions is called. However, when full module name is used (unaliased module name) for the macro, LFE compiler correctly recognizes it as a macro and expands it.

LFE 2.1.2 on OTP-26.0.2

**Comment by rvirding** [comment, marked as answer]
The reason for this is that the handling of aliases is done after all macro expansion.
